src/main/java/com/thevoxelbox/voxelsniper/brush/GenerateTreeBrush.java

```diff
@@ -116,71 +106,36 @@ private void leafNodeCreate() {
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY - y, blockPositionZ + z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY - y, blockPositionZ + z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX + x, blockPositionY - y, blockPositionZ + z));
-                                }
-                                this.clampY(blockPositionX + x, blockPositionY - y, blockPositionZ + z).setBlockData(leavesMaterial.createBlockData(), false);
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY - y, blockPositionZ - z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY - y, blockPositionZ - z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX + x, blockPositionY - y, blockPositionZ - z));
-                                }
-                                this.clampY(blockPositionX + x, blockPositionY - y, blockPositionZ - z).setBlockData(leavesMaterial.createBlockData(), false);
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX - x, blockPositionY - y, blockPositionZ + z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX - x, blockPositionY - y, blockPositionZ + z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX - x, blockPositionY - y, blockPositionZ + z));
-                                }
-                                this.clampY(blockPositionX - x, blockPositionY - y, blockPositionZ + z).setBlockData(leavesMaterial.createBlockData(), false);
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX - x, blockPositionY - y, blockPositionZ - z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX - x, blockPositionY - y, blockPositionZ - z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX - x, blockPositionY - y, blockPositionZ - z));
+                        for (int i = 0; i < 7; i++) {
+                            if (this.chance(30)) {
+                                switch (i) {
```

Comment:
- This is already a lot better then it was, but could we perhaps do it with three for loops instead? (like how I did it [here](https://github.com/KevinDaGame/VoxelSniper/blob/master/src/main/java/com/thevoxelbox/voxelsniper/brush/DrainBrush.java#L43-L45)).
- Also, createLeaf(x, y, z); is missing, I think 😉